docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,263 @@
+
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(column_data):
+        base = 500
+        return column_data / base
+
+Similarly we defined ``reverse_transform`` in a way that it operates on the data of a 
+single column.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(column_data):
+        base = 500
+        return column_data.round() * base
+
+Now that we have our functions, we initialize ``CustomConstraint`` and we 
+specify which column(s) are the desired ones.
+
+.. ipython:: python
+    :okwarning:
+
+    constraint = CustomConstraint(
+        columns=['salary', 'annual_bonus'],
+        transform=transform, 
+        reverse_transform=reverse_transform
+    )
+
+Now we create our model and pass our constraints.
+
+.. ipython:: python
+    :okwarning:
+
+    gc = GaussianCopula(constraints=[constraint])
+
+    gc.fit(employees)
+
+    sampled = gc.sample(10)
+
+Viewing ``sampled`` we now see two columns that are always a multiple of 500.
+
+.. ipython:: python
+    :okwarning:
+
+    sampled
+
+
+Can I access the rest of the table from my column functions?
+------------------------------------------------------------
+
+If we look closely at the data, we notice that ``salary`` and ``annual_bonus`` are only a 
+multiple of 500 when the employee is not a "contractor". To take this requirement into 
+consideration, we refer to a "fixed" column ``contractor`` in order to know whether we
+should apply this constraint or not. The access to ``contractor`` column will allow us
+to properly transform and reverse transform the data.
+
+We write our functions to take as input:
+
+-  ``table_data`` which contains all the information.
+-  ``column`` which is a an argument to represent the columns of interest.
+
+Now we can construct our functions freely, we write our methods
+with said arguments and be able to access ``'contractor'``.
+
+We first write our ``transform`` function as we have done previously:
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(table_data, column):
+        base = 500
+        display(table_data)
+        table_data[column] = table_data[column] / base
+        return table_data
+
+When it comes to defining ``reverse_transform``, we need to distinguish between
+contractors and non contractors, the operations are as follows:
+
+1. round values to four decimal points for contractors such that the end result will 
+   be two decimal points after multiplying the result with 500.
+2. round values to zero for employees that are not contractors such that the end
+   result will be a multiple of 500.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(table_data, column):
+        base = 500.
+        is_not_contractor = table_data.contractor == 0.
+        table_data[column] = table_data[column].round(4)
+        table_data.at[is_not_contractor, column] = table_data[column].loc[is_not_contractor].round()
```

Comment:
I think that `at` would not work here, since `at` [works for single values only](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.at.html#pandas.DataFrame.at). We need to use `loc` instead. Possibly irrelevant if we change to the proposal above.

Reply:
Is this warning ok to ignore?
```
SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame
```

I changed the above line to use `loc` and it seems to operate on the actual dataframe. So I am not sure why the warning is raised.

I also want to mention that `at` works for multiple values, I guess it's not the correct way though. We should revisit the change I made for GT to use `at` as well
https://github.com/sdv-dev/SDV/blob/cc6a9e0f4f8038f9a587a20c37f47e6f69eae07d/sdv/constraints/tabular.py#L464

Reply:
Yes, I think we should change it. Technically, the docs say that `at` is for single values, so we could expect this to change at any point.